core/controllers/beam_jobs.py

```diff
@@ -0,0 +1,111 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers responsible for managing Apache Beam jobs."""
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from core.controllers import acl_decorators
+from core.controllers import base
+from core.domain import beam_job_services
+import feconf
+from jobs import jobs_manager
+
+
+class BeamJobHandler(base.BaseHandler):
+    """Handler for getting the definitions of Apache Beam jobs."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {}
+    }
+
+    @acl_decorators.can_run_any_job
+    def get(self):
+        # type: () -> None
+        self.render_json({
+            'jobs': [j.to_dict() for j in beam_job_services.get_beam_jobs()]
+        })
+
+
+class BeamJobRunHandler(base.BaseHandler):
+    """Handler for managing the execution of Apache Beam jobs."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'job_name': {
+                'schema': {
+                    'type': 'unicode'
+                }
+            },
+            'job_arguments': {
+                'schema': {
+                    'type': 'list',
+                    'items': {
+                        'type': 'unicode'
+                    },
+                }
+            },
+        },
+    }
+
+    @acl_decorators.can_run_any_job
+    def get(self):
+        # type: () -> None
+        sorted_beam_job_runs = sorted(
+            beam_job_services.get_beam_job_runs(),
+            key=lambda j: j.job_updated_on,
+            reverse=True)
+
+        self.render_json({
+            'runs': [r.to_dict() for r in sorted_beam_job_runs]
+        })
+
+    @acl_decorators.can_run_any_job
+    def put(self):
+        # type: () -> None
+        job_name = self.payload.get('job_name')
+        job_args = self.payload.get('job_arguments')
```

Comment:
Please do
```suggestion
        job_name = self.normalized_payload.get('job_name')
        job_args = self.normalized_payload.get('job_arguments')
```
cc @Nik-09 @rohitkatlaa should we make sure that in the controller files (except for base) devs actually use the normalized payload and normalized request?

Reply:
True, we should use normalized_payload for them.

Reply:
@Nik-09 Should we have some kind of check for this? Like linter?

Reply:
Yes, it will be better to have one.
I am appending -- ```Add linters to block use of payload in the handler``` in the doc that I am preparing for additional features for schema validation.

Reply:
Done.